x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,518 @@
+        streams:
+          - dataset: foo.dataset
+            topic: foo
+            processors:
+             - extract_bro_specifics
+
+
+  #################################################################################################
+  # System EPM package
+  - id?: system
+    title: Collect system information and metrics
+    package:
+      name: epm/system
+      version: 1.7.0
+    inputs:
+      - type: metrics/system
+        streams:
+         - id?: {id}
+           enabled?: false # default true
+           metricset: cpu
+           period: 10s
+           dataset: system.cpu
+           metrics: ["percentages", "normalized_percentages"]
+         - metricset: memory
+           dataset: system.memory
+         - metricset: diskio
+           dataset: system.diskio
+         - metricset: load
+           dataset: system.load
+         - metricset: memory
+           dataset: system.memory
+         - metricset: process
+           dataset: system.process
+           processes: ["firefox*"]
+         - metricset: process_summary
+           dataset: system.process_summary
+         - metricset: uptime
+           dataset: system.uptime
+         - metricset: socket_summary
+           dataset: system.socket_summary
+         - metricset: filesystem
+           dataset: system.filesystem
+         - metricset: raid
+           dataset: system.raid
+         - metricset: socket
+           dataset: system.socket
+         - metricset: service
+           dataset: system.service
+         - metricset: fsstat
+           dataset: system.fsstat
+         - metricset: foo
+           dataset: system.foo
+
+
+  #################################################################################################
+  # Elasticsearch package example
+  - id?: my-endpoint
+    title: Collect Elasticsearch information
+    package:
+      name: epm/elsticsearch
+      version: 1.7.0
+    inputs:
+        - type: log
+          streams:
+          - id?: {id}
+            enabled?: true # default to true
+            dataset: elasticsearch.audit
+            paths: [/var/log/elasticsearch/*_access.log, /var/log/elasticsearch/*_audit.log]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.deprecation
+            paths: [/var/log/elasticsearch/*_deprecation.log]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.gc
+            paths: [/var/log/elasticsearch/*_gc.log, /var/log/elasticsearch/*_gc.log.[0-9]*]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.server
+            paths: [/var/log/elasticsearch/*.log]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.slowlog
+            paths: [/var/log/elasticsearch/*_index_search_slowlog.log, /var/log/elasticsearch/*_index_indexing_slowlog.log]
+        - type: metrics/elasticsearch
+          hosts: ["http://localhost:9200"]
```

Comment:
Can you add a few more common configs here like username, password, period, ssl.cerficate_authorities to make it more obvious the common configs are here? https://github.com/elastic/beats/blob/master/metricbeat/module/elasticsearch/_meta/config.reference.yml#L12

Reply:
@ruflin I thought period was at the stream level and not on the common part of the input?

Reply:
It is, skip it for now

We will probably need to come back to the question: Are there configs that can be specified on both levels like period as default and have a local period that overwrites it. Lets have this discussion later.